runtime/src/main/java/org/corfudb/runtime/collections/TxBuilder.java

```diff
@@ -100,6 +101,37 @@ TxBuilder update(@Nonnull final String tableName,
         return this;
     }
 
+    /**
+     * Updates the value on the specified key by applying a BiFunction
+     *
+     * @param tableName Table name to perform the update on.
+     * @param key       Key
+     * @param mutation  Function to apply to get the new value
+     * @param arg       Argument to pass to the mutation function
+     * @param <K>       Type of Key.
+     * @param <V>       Type of Value.
+     * @return TxBuilder instance
+     */
+    @Nonnull
+    public <K extends Message, V extends Message, M extends Message>
+    TxBuilder update(@Nonnull final String tableName,
```

Comment:
What is done here is interesting, it looks like the caller can actually specify a delta update and it is up to the database to apply the update.
Right now you have implemented it as a classic read-modify-write cycle. But in the future we could have corfu defer the application of the 'arg' or delta here and lazily apply them (say at compaction). This is a powerful feature provided by other databases and if the goal is to accomplish this functionality perhaps we can just rename this function to call it 
merge(), the Bifunction as the mergeOperator and the arg as the delta?
Later we can think of optimizing this in corfu to support lazy merge to defer the read in the read-modify-write cycle.